tgui/packages/tgui/interfaces/PaiInterface.tsx

```diff
@@ -0,0 +1,540 @@
+        <Button
+          icon="download"
+          onClick={() => act(software.toLowerCase().replace(/ /g, '_'))}
+          disabled={!!languages}>
+          {!languages ? 'Install' : 'Installed'}
+        </Button>
+      );
+    default:
+      return (
+        <Button
+          icon="power-off"
+          onClick={() => act(software.toLowerCase().replace(/ /g, '_'))}
+          tooltip="Attempts to toggle the module's power.">
+          Toggle
+        </Button>
+      );
+  }
+};
+
+const AvailableDisplay = () => {
+  return (
+    <Section
+      buttons={<AvailableMemory />}
+      fill
+      scrollable
+      title="Available Software">
+      <AvailableSoftware />
+    </Section>
+  );
+};
+
+const AvailableMemory = (_, context) => {
+  const { data } = useBackend<PaiInterfaceData>(context);
+  const { ram } = data;
+
+  return (
+    <Tooltip content="Available System Memory">
+      <Stack>
+        <Icon color="purple" mt={0.7} mr={1} name="microchip" />
+        <ProgressBar
+          minValue={0}
+          maxValue={100}
+          ranges={{
+            good: [67, 100],
+            average: [34, 66],
+            bad: [0, 33],
+          }}
+          value={ram}
+        />
+      </Stack>
+    </Tooltip>
+  );
+};
+
+const AvailableSoftware = (_, context) => {
+  const { act, data } = useBackend<PaiInterfaceData>(context);
+  const { available, installed } = data.software;
+  /** Converted to type: Available */
+  let convertedList: Available[] = Object.keys(available).map((key) => {
+    return {
+      name: key,
+      value: available[key],
+    };
+  });
+
+  return (
+    <Table>
+      {convertedList.map((software) => {
+        return <AvailableRow key={software.name} software={software} />;
+      })}
+    </Table>
+  );
+};
+
+const AvailableRow = (props, context) => {
+  const { act, data } = useBackend<PaiInterfaceData>(context);
+  const { ram } = data;
+  const { installed } = data.software;
+  const { software } = props;
+  const purchased = installed.includes(software.name);
+
+  return (
+    <Table.Row>
+      <Table.Cell>
+        <Box color="label">
+          {software.name.replace(/^\w/, (c) => c.toUpperCase())}
+        </Box>
+      </Table.Cell>
+      <Table.Cell width={8} />
+      <Table.Cell>
+        <Box color={ram < software.value && 'bad'} textAlign="right">
+          {!purchased && software.value}{' '}
+          <Icon
+            color={purchased || !(ram < software.value) ? 'purple' : 'bad'}
+            name="microchip"
+          />
+        </Box>
+      </Table.Cell>
+
+      <Table.Cell>
+        <Button
+          disabled={ram < software.value || purchased}
+          onClick={() => act('buy', { selection: software.name })}
+          tooltip={<SoftwareInfo software={software.name} />}>
+          <Icon ml={0.7} name="download" />
+        </Button>
+      </Table.Cell>
+    </Table.Row>
+  );
+};
+
+const getIconHelper = (image: string) => {
+  switch (image) {
```

Comment:
This one doesn't matter so much but I still think they should be a const object.

Reply:
Will do